pylint/checkers/classes/class_checker.py

```diff
@@ -904,6 +911,33 @@ def _check_proper_bases(self, node: nodes.ClassDef) -> None:
                         confidence=INFERENCE,
                     )
 
+            if isinstance(ancestor, nodes.ClassDef) and ancestor.is_subtype_of(
+                "enum.IntFlag"
+            ):
+                previous_values, total, union, overlaps = set(), None, None, []
+                for assign_name in node.nodes_of_class(nodes.AssignName):
+                    if not isinstance(assign_name.parent, nodes.Assign):
+                        continue  # Ignore non-assignment expressions
+
+                    assigned = assign_name.parent.value
+                    if not isinstance(assigned, nodes.Const):
+                        continue  # Ignore non-literal assignments
+
+                    assert assigned.value is not None
+                    if assigned.value in previous_values:
+                        continue  # Ignore aliases
+                    previous_values.add(assigned.value)
+
+                    # Detect divergence between the sum-of-values and union-of-values
+                    total = assigned.value if total is None else total + assigned.value
+                    union = assigned.value if union is None else union | assigned.value
+                    if total != union:
+                        overlaps.append(assign_name)
+                        total = union  # reset divergence detection after each iteration
+
+                for overlap in overlaps:
+                    self.add_message("implicit-flag-overlap", node=overlap)
```

Comment:
add_message should also have `confidence=INFERENCE` (and update functional test .txt output accordingly)